src/AppInstallerCLICore/Argument.cpp

```diff
@@ -108,6 +109,19 @@ namespace AppInstaller::CLI
         args.push_back(ForType(Args::Type::VerboseLogs));
     }
 
+    void Argument::ValidatePackageSelectionArgumentSupplied(const Execution::Args& args)
+    {
+        for (Args::Type type : { Args::Type::Query, Args::Type::Manifest, Args::Type::Id, Args::Type::Name, Args::Type::Moniker, Args::Type::Tag, Args::Type::Command })
+        {
+            if (args.Contains(type))
+            {
+                return;

```

Comment:
Would it be possible to ensure the query is also not an empty string?
https://github.com/microsoft/winget-cli/issues/1249

Reply:
It would be possible, but allowing an empty string does allow for an effective `--all` being `""`.  The current change prevents accidental "everything" searches, which I feel was the primary value.  Preventing intentional searching for everything seems like it would be due to the current performance limitations rather than due to protecting users.

If the user types in `winget search ""`, it could be an accident, but it is more likely that it is intentional than `winget search`.  I could see scripts constructing the command line resulting in more frequent empty strings as well, but if the script author wants to block that, they can.

Without any other reasons, I'm not inclined to block the empty strings.

Reply:
I think those are fair points, and are things I hadn't considered. After thinking more, I do agree that simply requiring an argument solves the potential issues here